### PR TITLE
Add multi-schema support to the json_stream_sampler

### PR DESCRIPTION
**Pull request by tom95858**
This change adds support for multiple schema to the json_stream_sampler.

### CODE REVIEW

**Review by baallan — CHANGES_REQUESTED** [review]
@tom95858 this is in the right direction, but it turns out the spank and netlink use cases need a schema_format (akin instance_format) option to work well. Details in the review.

**Comment by baallan** [comment]
Man page needs updating once the changes are settled.

File: ldms/src/sampler/json/json_stream_sampler.c [hunk 1]
```diff
@@ -104,26 +105,52 @@ struct attr_entry {
...
+	pthread_mutex_t lock;
 };
-static struct rbt schema_tree = RBT_INITIALIZER(str_cmp);
-static pthread_mutex_t schema_tree_lock = PTHREAD_MUTEX_INITIALIZER;
 
 static const char *usage(struct ldmsd_plugin *self)
 {
 	return \
-	"config name=json_stream_sampler producer=<producer_name> \n"
+	"config name=js_stream_sampler producer=<prod_name> \n"
```
[the remark this answers is not in the document]
Comment (tom95858) [reply ldms/src/sampler/json/json_stream_sampler.c:151]: @baallan, I think you are correct.

**Comment by emdonat** [comment]
@tom95858 You mentioned last week that a fix for this could be done easily.  Can you update status?

**Pull request merged**